utilities/write_batch_with_index/write_batch_with_index.cc

```diff
@@ -216,9 +218,19 @@ class BaseDeltaIterator : public Iterator {
     }
     // equal_keys_ <=> compare == 0
     assert((equal_keys_ || compare != 0) && (!equal_keys_ || compare == 0));
+
 #endif
   }
 
+  void IsOverUpperBound() {
+    if ((read_options_.iterate_upper_bound != nullptr &&
+         comparator_->Compare(key(),
+                              *read_options_.iterate_upper_bound) >= 0)) {
```

Comment:
A slightly more efficient way is to filter out delta iterator, rather than merged result. This allows us to skip lots of tombstones in write batch.